wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt

```diff
@@ -201,26 +201,29 @@ actual abstract class ProtoAdapter<E> actual constructor(
 
     // Obsolete; for Java classes generated before typeUrl and syntax were added.
     @JvmStatic fun <M : Message<M, B>, B : Message.Builder<M, B>> newMessageAdapter(
-      type: Class<M>
+      type: Class<M>,
+      loader: ClassLoader,
     ): ProtoAdapter<M> {
-      return createRuntimeMessageAdapter(type, null, Syntax.PROTO_2)
+      return createRuntimeMessageAdapter(type, null, Syntax.PROTO_2, loader)
     }
 
     // Obsolete; for Java classes generated before typeUrl and syntax were added.
     @JvmStatic fun <M : Message<M, B>, B : Message.Builder<M, B>> newMessageAdapter(
       type: Class<M>,
-      typeUrl: String
+      typeUrl: String,
+      loader: ClassLoader,
     ): ProtoAdapter<M> {
-      return createRuntimeMessageAdapter(type, typeUrl, Syntax.PROTO_2)
+      return createRuntimeMessageAdapter(type, typeUrl, Syntax.PROTO_2, loader)
     }
 
     /** Creates a new proto adapter for `type`. */
     @JvmStatic fun <M : Message<M, B>, B : Message.Builder<M, B>> newMessageAdapter(
```

Comment:
We keep this method, update the comment saying it's obsolete but we keep it for classes generated before `loader` was added. and call the newly created method with the default loader (I think `ProtoAdapter::class.java.classLoader`?)